packages/@lwc/integration-karma/test/light-dom/slotting/index.spec.js

```diff
@@ -119,6 +121,39 @@ describe('Slotting', () => {
         );
     });
 
+    it('should forward slots between shadow and light DOM', () => {
+        const nodes = createTestElement(
+            'x-forwarded-slot-consumer-shadow',
+            forwardedSlotConsumerShadow
+        );
+        const elm = nodes['x-forwarded-slot-consumer-shadow'];
+        expect(elm.innerHTML).toEqual(
+            process.env.NATIVE_SHADOW // native shadow doesn't output slots in innerHTML
+                ? // innerHTML in native shadow only renders the order of the slotted content but not the slot order
+                  `<x-forward-light-slot-to-shadow><x-shadow-container><p slot="lower">Upper slot content forwarded</p><p>Default slot forwarded</p><p slot="upper">Lower slot content forwarded</p></x-shadow-container></x-forward-light-slot-to-shadow>`
+                : `<x-forward-light-slot-to-shadow><x-shadow-container><slot name="upper" data-id="upper-slot"><p slot="upper">Lower slot content forwarded</p></slot><slot data-id="default-slot"><p>Default slot forwarded</p></slot><slot name="lower" data-id="lower-slot"><p slot="lower">Upper slot content forwarded</p></slot></x-shadow-container></x-forward-light-slot-to-shadow>`
+        );
+
+        if (process.env.NATIVE_SHADOW) {
```

Comment:
I recommend avoiding `innerHTML` entirely since it's buggy between native/synthetic shadow. Ideally we should have the same tests here for both native and synthetic.